### PR TITLE
Show last command and running commands in title with shell_integration

### LINKED ISSUE #10260: Set terminal title to current command with shell_integration
### Related problem

When I start a long-running or interactive process (e.g. `du -sh /*` or `vim`), zsh sets the window title to the command. That's quite convenient, especially for interactive commands, to tell multiple windows / tmux tabs apart. But nushell doesn't do this; the title is always the working directory (with `shell_integration` enabled).

### Describe the solution you'd like

Nushell should automatically set the window title to the currently-running command, if any. I imagine not everyone likes this, so this should probably be configurable, specifically:
- Add a config option, `set_title_to_running_command` or so, that is only relevant if `shell_integration = true`.
- If `set_title_to_running_command = true` and a command is executed, the window title is set to the text of that command. When the command finishes, the window title is reset to the current directory (or whatever it would be normally within nu).

### Describe alternatives you've considered

I guess I could use a hook based on the alternative solution in #6388 to achieve this. But it only works with `shell_integration=false` and I'm not sure how stable it'd be. Also a bit inconvenient of course.

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Closes #10260

I'm not 100% convinced about the star thing for running commands because on short commands it makes the title jitter.

### CODE REVIEW

**Comment by fdncred** [comment]
What do other shells do like zsh or fish? I don't really want a flashing title bar.

**Comment by SUPERCILEX** [comment]
No clue unfortunately as I haven't used those. I'd be happy to just show the last executed command if that's preferable.

**Comment by fdncred** [comment]
This is what Ghostty looks like in fish.
image
I was in `~/s/nushell` and my command was `cargo install --path . --features=extra,dataframe` so it looks like fish is truncating at around 20 chars and showing the path.

And then when the install finished it returned to this.
image

**Comment by fdncred** [comment]
I do ls and it briefly flashes `ls ~/s/nushell` and then back to `/s/nushell`

**Comment by SUPERCILEX** [comment]
Hmmm, so I guess flashing is standard then? Definitely don't like that the path is at the end since it looks like it's an arg to the command. We could do truncation, but that's also very questionable since you don't know what you're truncating.

**Comment by fdncred** [comment]
I like how it resets to the just the path without having the command. so, if you're running a command, you see it, but when it finishes it's just the path.

**Comment by SUPERCILEX** [comment]
~~I actually kind of like it the other way: then you can easily see the context for that tab/window since you know the last command that was run.~~

I thought about this some more and I take it back. A terminal can be approximated as isomorphic with respect to the CWD (modulo the environment), so actually any terminal with a given directory should be fungible and therefore the directory (and not the last command) is the context you'd use to identify the terminal.

TL;DR: cool, I'll do it that way.

**Comment by fdncred** [comment]
ok, thanks. let's give this a try.
